addons/sale/tests/test_sale_pricelist.py

```diff
@@ -1,132 +1,120 @@
-            'price_unit': cls.product_order.list_price,
+            'product_uom': cls.company_data['product_order_no'].uom_id.id,
+            'price_unit': cls.company_data['product_order_no'].list_price,
             'order_id': cls.sale_order.id,
             'tax_id': False,
         })
         cls.sol_serv_deliver = SaleOrderLine.create({
-            'name': cls.service_deliver.name,
-            'product_id': cls.service_deliver.id,
+            'name': cls.company_data['product_service_delivery'].name,
+            'product_id': cls.company_data['product_service_delivery'].id,
             'product_uom_qty': 2,
-            'product_uom': cls.service_deliver.uom_id.id,
-            'price_unit': cls.service_deliver.list_price,
+            'product_uom': cls.company_data['product_service_delivery'].uom_id.id,
+            'price_unit': cls.company_data['product_service_delivery'].list_price,
             'order_id': cls.sale_order.id,
             'tax_id': False,
         })
         cls.sol_serv_order = SaleOrderLine.create({
-            'name': cls.service_order.name,
-            'product_id': cls.service_order.id,
+            'name': cls.company_data['product_service_order'].name,
+            'product_id': cls.company_data['product_service_order'].id,
             'product_uom_qty': 2,
-            'product_uom': cls.service_order.uom_id.id,
-            'price_unit': cls.service_order.list_price,
+            'product_uom': cls.company_data['product_service_order'].uom_id.id,
+            'price_unit': cls.company_data['product_service_order'].list_price,
             'order_id': cls.sale_order.id,
             'tax_id': False,
         })
         cls.sol_prod_deliver = SaleOrderLine.create({
-            'name': cls.product_deliver.name,
-            'product_id': cls.product_deliver.id,
+            'name': cls.company_data['product_delivery_no'].name,
+            'product_id': cls.company_data['product_delivery_no'].id,
             'product_uom_qty': 2,
-            'product_uom': cls.product_deliver.uom_id.id,
-            'price_unit': cls.product_deliver.list_price,
+            'product_uom': cls.company_data['product_delivery_no'].uom_id.id,
+            'price_unit': cls.company_data['product_delivery_no'].list_price,
             'order_id': cls.sale_order.id,
             'tax_id': False,
         })
 
-    def test_sale_with_pricelist_empty(self):
-        # Trigger onchange to reset discount, unit price, subtotal, ...
-        self.sale_order.onchange_partner_id()
-        for line in self.sale_order.order_line:
-            line.product_id_change()
-            line._onchange_discount()
-        # Check that pricelist of the SO has been applied on the sale order lines or not
-        for line in self.sale_order.order_line:
-            self.assertEqual(line.price_unit, line.product_id.list_price, 'Pricelist of the SO should not be applied on an order line %s' % (line.name,))
-
```

Comment:
no substitute?

Reply:
this test is already tested somewhere else in this file and was not-deterministic